Android linking fails with NDK r11b: undefined reference to 'bsd_signal'

Cross-compiling on Linux x64 to Android, rustc compiled from nightly sources as of 2016-03-17.

$ rustc --version -v
rustc 1.9.0-dev
binary: rustc
commit-hash: unknown
commit-date: unknown
host: x86_64-unknown-linux-gnu
release: 1.9.0-dev

$ cat - >hello.rs
fn main() {
  println!("Hello, rusty Android!");
}
^D

$ rustc --target=arm-linux-androideabi -C linker=$HOME/Work/Android/NdkStandalone/bin/arm-linux-androideabi-gcc -C link-args="-pie -fPIE" hello.rs
error: linking with `/home/skirmantas/Work/Android/NdkStandalone/bin/arm-linux-androideabi-gcc` failed: exit code: 1
note: "/home/skirmantas/Work/Android/NdkStandalone/bin/arm-linux-androideabi-gcc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-Wl,--allow-multiple-definition" "-L" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib" "hello.0.o" "-o" "hello" "-Wl,--gc-sections" "-pie" "-nodefaultlibs" "-L" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib" "-Wl,-Bstatic" "-Wl,-Bdynamic" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/libstd-9026086f.rlib" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/libcollections-9026086f.rlib" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/librustc_unicode-9026086f.rlib" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/librand-9026086f.rlib" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/liballoc-9026086f.rlib" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/liballoc_jemalloc-9026086f.rlib" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/liblibc-9026086f.rlib" "/usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/libcore-9026086f.rlib" "-l" "dl" "-l" "log" "-l" "gcc" "-l" "gcc" "-l" "c" "-l" "m" "-pie" "-fPIE" "-l" "compiler-rt"
note: /usr/local/stow/rustc-20160317/lib/rustlib/arm-linux-androideabi/lib/libstd-9026086f.rlib(std-9026086f.0.o):std.0.rs:function rt::lang_start::h1deaaba938bd63a9Xqz: error: undefined reference to 'bsd_signal'
collect2: error: ld returned 1 exit status

error: aborting due to previous error

--- a/libc-test/build.rs
+++ b/libc-test/build.rs
@@ -385,12 +385,6 @@ fn main() {
 
     cfg.skip_fn_ptrcheck(move |name| {
         match name {
-            // This used to be called bsd_signal in rev 18 of the android
-            // platform and is now just called signal, the old `bsd_signal`
-            // symbol, however, still remains, just gives a different function
-            // pointer.
-            "signal" if android => true,
-
             // dllimport weirdness?
             _ if windows => true,
 

--- a/src/unix/mod.rs
+++ b/src/unix/mod.rs
@@ -408,7 +408,6 @@ extern {
 
     pub fn ftruncate(fd: ::c_int, length: off_t) -> ::c_int;
 
-    #[cfg_attr(target_os = "android", link_name = "bsd_signal")]
     pub fn signal(signum: ::c_int, handler: sighandler_t) -> sighandler_t;
 
     #[cfg_attr(all(target_os = "macos", target_arch = "x86"),

